### PR TITLE
Androapp 4930

### PR DESCRIPTION
**Pull request by mvmmacamo2**
## Description
Replace Sync button behaviour

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-4930

## Solution description
1. Install hilt worker dependence 
2. Delete Settings activity
3. Create provides HiltWorkerFactory
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [x ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [x ] Other
## Checklist
- [ x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeActivity.java [hunk 1]
```diff
@@ -93,12 +99,62 @@ private void synchronizeData() {
         binding.syncActionButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                Intent intent = new Intent(HomeActivity.this, SettingsActivity.class);
-                startActivity(intent);
+
+//                Intent intent = new Intent(HomeActivity.this, SettingsActivity.class);
+//                startActivity(intent);
```
Comment (Balcan) [inline stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeActivity.java:104]: Remove commented code if not needed

**Review by Balcan — APPROVED** [review]

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeActivity.java (hunk 1 above)
Comment (mvmmacamo2) [reply stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeActivity.java]: Done. Thank you, @Balcan.

File: buildsystem/dependencies.gradle [hunk 2]
```diff
@@ -123,6 +123,8 @@ ext {
             hilt                        : '2.43.2',
             hilt_work                   : '1.0.0',
 
+            workmanager                   : '2.7.1',
```
Comment (andresmr) [inline buildsystem/dependencies.gradle:126]: are you using this dependency?

File: app/src/main/java/org/dhis2/AppComponent.java [hunk 3]
```diff
@@ -1,5 +1,8 @@
 package org.dhis2;
 
+
+import androidx.hilt.work.HiltWorkerFactory;
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/AppComponent.java:4]: are you using this?

**Review by andresmr — COMMENTED** [review]
Very good pull request, I add a couple of comments of code that i don't think it is used.

File: buildsystem/dependencies.gradle (hunk 2 above)
Comment (mvmmacamo2) [reply buildsystem/dependencies.gradle]: No, I forgot to remove, but I have removed.

File: app/src/main/java/org/dhis2/AppComponent.java (hunk 3 above)
Comment (mvmmacamo2) [reply app/src/main/java/org/dhis2/AppComponent.java]: No. I have removed.

**Review by andresmr — APPROVED** [review]

**Pull request merged**